xen/drivers/passthrough/iommu.c

```diff
@@ -249,22 +249,39 @@ void iommu_domain_destroy(struct domain *d)
     arch_iommu_domain_destroy(d);
 }
 
-int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
-                   unsigned int flags)
+int iommu_map_pages(struct domain *d, unsigned long gfn, unsigned long mfn,
+                    unsigned long page_count, unsigned int flags)
 {
     const struct domain_iommu *hd = dom_iommu(d);
-    int rc;
+    int rc = 0;
+    unsigned long i;
 
     if ( !iommu_enabled || !hd->platform_ops )
         return 0;
 
-    rc = hd->platform_ops->map_page(d, gfn, mfn, flags);
+    if ( hd->platform_ops->map_pages &&
+         (page_count > 1 || !hd->platform_ops->map_page) )
+        rc = hd->platform_ops->map_pages(d, gfn, mfn, page_count, flags);
```

Comment:
what is the point of checking if page_count is 1? It can anyways be handled in the loop with the same hd->platform_ops->map_page